infra/Pos/Discovery/Holders.hs

```diff
@@ -112,27 +118,23 @@ data DiscoveryContextSum
 -- uses only one of them).
 type MonadDiscoverySum = Ether.MonadReader' DiscoveryContextSum
 
-data DiscoveryRedirectTag
-
-type DiscoveryRedirect =
-    Ether.TaggedTrans DiscoveryRedirectTag IdentityT
-
-runDiscoveryRedirect :: DiscoveryRedirect m a -> m a
-runDiscoveryRedirect = coerce
-
 askDiscoveryContextSum :: MonadDiscoverySum m => m DiscoveryContextSum
 askDiscoveryContextSum = Ether.ask'
 
-instance (MonadDiscoverySum m, DiscoveryKademliaEnv m, t ~ IdentityT) =>
-         MonadDiscovery (Ether.TaggedTrans DiscoveryRedirectTag t m) where
-    getPeers =
-        Ether.ask' >>= \case
-            DCStatic nodes -> runDiscoveryConstT nodes getPeers
-            DCKademlia inst -> runDiscoveryKademliaT inst getPeers
-    findPeers =
-        Ether.ask' >>= \case
-            DCStatic nodes -> runDiscoveryConstT nodes findPeers
-            DCKademlia inst -> runDiscoveryKademliaT inst findPeers
+type DiscoveryRealMonad m =
+    (MonadDiscoverySum m, DiscoveryKademliaEnv m)
+
+getPeersReal :: DiscoveryRealMonad m => m (Set NodeId)
+getPeersReal =
+    Ether.ask' >>= \case
+        DCStatic nodes -> runDiscoveryConstT nodes getPeers
+        DCKademlia inst -> runDiscoveryKademliaT inst getPeers
+
+findPeersReal :: DiscoveryRealMonad m => m (Set NodeId)
+findPeersReal =
```

Comment:
I don't like suffix `Real` in this module. `DiscoveryKademlia` and `DiscoveryConstT` are as real as their sum. Why not change `Real` to `Sum`. What does this `Real` actually mean?